docs/dev/developer-guides/building-on-zksync/events.md

```diff
@@ -14,30 +14,50 @@ At zkSync, events behave the same way as in Ethereum.
 ## Events filtering
 
 Filtering is used to query indexed data and provide lower-cost data storage when the data is not required to be accessed on-chain.
+While filtering, we advice to load events by block ranges (0-1999, 2000-3999, ...) and index the result on your end, else you will get an error that says "block range should be less than or equal to 2000".
 
 These can be used in conjunction with the [Provider Events API](https://docs.ethers.io/v5/api/providers/provider/#Provider--event-methods) and with the [Contract Events API](https://docs.ethers.io/v5/api/contract/contract/#Contract--events).
 
+## Getting the events
 
+To get the events, use the following template:
 
-```solidity
-    abi = [
-  "event Transfer(address indexed src, address indexed dst, uint val)"
-];
+```js
+const ether = require("ethers");
+const contractABI = require("./ABI_JSON")
 
-contract = new Contract(tokenAddress, abi, provider);
+const getEvent = async () => {
+const contractAddress = "<YOUR ADDRESS>"
+const provider = new ether.providers.WebSocketProvider(`wss://zksync2-testnet.zksync.dev/ws`)
 
-// List all token transfers *to* myAddress:
-contract.filters.Transfer(null, myAddress)
-//{
-//    "jsonrpc":"2.0",
-//   "id":1,
-//    "method":"eth_getLogs",
-//    "params":[{
-//       "fromBlock": "0x8e8e6",
-//        "toBlock": "0x8e8e6",
-//        "topics": [["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"]]
-//    }]
-//}
+const contract = new ether.ethers.Contract(contractAddress, contractABI, provider);
+
+await contract.on("Transfer", (event) => {
+    let info = {
+        filter: { INDEXED_PARAMETER: VALUE },
+        fromBlock: BLOCK_NUMBER, // e.g 15943000
+        toBlock: BLOCK_NUMBER, // e.g 15943100
+        data: event,
+    }
+    console.log(JSON.stringify(info, null, 4))
+})
+}
+
+getEvent()
 
 ```
-zkSync has a 10K log limit per response.
+
+where
+
+Provider — Your websocket provider through which you will retrieve the events data.
+Contract address — The contract address whose events you want to track.
+ABI — The ABI(Application Binary Interface) of the contract in JSON format.
+Transfer — Any name you want to assign to print it at the end, for this sample, we used "Transfer".
+Event name — The name of the event as defined in the smart contract.
```

Comment:
Where is event name in the code?

Reply:
fixed in 79e18fb8c9dd426d700ab809eaeff8251ccb799e